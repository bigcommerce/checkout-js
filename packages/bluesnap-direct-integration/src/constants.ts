export enum BluesnapECPAccountType {
    ConsumerChecking = 'CONSUMER_CHECKING',
    ConsumerSavings = 'CONSUMER_SAVINGS',
    CorporateChecking = 'CORPORATE_CHECKING',
    CorporateSavings = 'CORPORATE_SAVINGS',
}
