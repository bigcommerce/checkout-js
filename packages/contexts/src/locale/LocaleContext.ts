import type { CurrencyService, LanguageService } from '@bigcommerce/checkout-sdk';
import { createContext } from 'react';

export interface LocaleContextType {
    language: LanguageService;
    date?: {
        inputFormat: string;
    };
    currency?: CurrencyService;
}

export const LocaleContext = createContext<LocaleContextType | undefined>(undefined);
