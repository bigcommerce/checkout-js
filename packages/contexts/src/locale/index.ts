export { LocaleContext, type LocaleContextType } from './LocaleContext';
export { default as LocaleProvider } from './LocaleProvider';
export { useLocale } from './useLocale';
