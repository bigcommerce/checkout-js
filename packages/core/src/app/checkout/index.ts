export { RenderCheckout, RenderCheckoutOptions } from './renderCheckout';
export { default as CheckoutContext, CheckoutContextProps } from './CheckoutContext';
export {
    default as CheckoutProvider,
    CheckoutProviderProps,
    CheckoutProviderState,
} from './CheckoutProvider';
export { default as withCheckout, WithCheckoutProps } from './withCheckout';
export { default as CheckoutSupport } from './CheckoutSupport';
export { default as NoopCheckoutSupport } from './NoopCheckoutSupport';
export { default as navigateToOrderConfirmation } from './navigateToOrderConfirmation';
