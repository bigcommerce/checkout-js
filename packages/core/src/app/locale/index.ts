export { default as LocaleContext, LocaleContextType } from './LocaleContext';
export { default as createLocaleContext } from './createLocaleContext';
export { default as getDefaultTranslations } from './getDefaultTranslations';
export { default as getLanguageService, InitializeLanguageService } from './getLanguageService';
export { default as masterpassFormatLocale } from './masterpassFormatLocale';
export { default as withCurrency, WithCurrencyProps } from './withCurrency';
export { default as withLanguage, WithLanguageProps } from './withLanguage';
export { default as withDate, WithDateProps } from './withDate';
export { default as LanguageWindow, isLanguageWindow } from './LanguageWindow';
export { default as LocaleProvider } from './LocaleProvider';
export { default as TranslatedHtml, TranslatedHtmlProps } from './TranslatedHtml';
export { default as TranslatedLink, TranslatedLinkProps } from './TranslatedLink';
export { default as TranslatedString, TranslatedStringProps } from './TranslatedString';
