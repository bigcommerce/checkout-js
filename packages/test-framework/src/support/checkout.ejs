<!DOCTYPE html>
<html>
    <head>
        <title>Local Testing Environment</title>
        <base href="/" target="_top"/>
        <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
        <link href="https://fonts.googleapis.com/css?family=Karla:400|Montserrat:400,500,700&display=swap" rel="stylesheet" />
    </head>
    <body>
    <header class="checkoutHeader optimizedCheckout-header">
        <div class="checkoutHeader-content">
            <h1 class="is-srOnly">Checkout</h1>
            <h2 class="checkoutHeader-heading">
                <a class="checkoutHeader-link" href="<%= storeUrl %>/">
                    <span class="header-logo-text">Local Testing Environment</span>
                </a>
            </h2>
        </div>
    </header>
    <div class="ng-checkout-container" id="micro-app-ng-checkout">
        <div class="ng-checkout-view" data-ui-view></div>
    </div>
    <script type="text/javascript">
        window.checkoutConfig = {
            checkoutId: '<%= checkoutId %>',
            containerId: 'micro-app-ng-checkout',
        };
    </script>
    <script src="<%= localhostUrl %>/auto-loader.js" type="text/javascript"></script>
    <script type="text/javascript">
        (function() {
            window.checkout.renderCheckout({
                checkoutId: '<%= checkoutId %>',
                containerId: 'micro-app-ng-checkout',
                sentryConfig: null,
            })
        })();
    </script>
    </body>
</html>
