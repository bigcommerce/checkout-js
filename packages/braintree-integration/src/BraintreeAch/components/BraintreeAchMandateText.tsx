import React, { FunctionComponent, useEffect } from 'react';

import { useLocale } from '@bigcommerce/checkout/locale';
import { useCheckout, usePaymentFormContext } from '@bigcommerce/checkout/payment-integration-api';
import { CheckboxFormField } from '@bigcommerce/checkout/ui';

import { OwnershipTypes } from '../constants';

export interface BraintreeAchMandateTextProps {
    isInstrumentFeatureAvailable?: boolean;
    updateMandateText: (mandateText: string) => void;
}

const BraintreeAchMandateText: FunctionComponent<BraintreeAchMandateTextProps> = ({
    isInstrumentFeatureAvailable,
    updateMandateText,
}) => {
    const { checkoutState } = useCheckout();
    const { paymentForm } = usePaymentFormContext();
    const { language } = useLocale();

    const {
        businessName,
        accountType,
        accountNumber,
        routingNumber,
        ownershipType,
        firstName,
        lastName,
    } = paymentForm.getFormValues();

    const fullName = `${String(firstName)} ${String(lastName)}`;

    const depositoryName = ownershipType === OwnershipTypes.Business ? businessName : fullName;
    const outstandingBalance = checkoutState.data.getCheckout()?.outstandingBalance;
    const storeName = checkoutState.data.getConfig()?.storeProfile.storeName;
    const symbol = checkoutState.data.getCart()?.currency.symbol;

    const mandateTextTranslationKey = isInstrumentFeatureAvailable
        ? 'payment.braintreeach_vaulting_mandate_text'
        : 'payment.braintreeach_mandate_text';

    const currentDate = new Date(Date.now()).toJSON().slice(0, 10); // yyyy-mm-dd
    const formattedDate = currentDate.split('-').reverse().join('.'); // dd.mm.yyyy

    const mandateTextTranslationProps = {
        accountType: String(accountType).toLowerCase(),
        accountNumber: String(accountNumber),
        currentDate: formattedDate,
        depositoryName: String(depositoryName),
        routingNumber: String(routingNumber),
        outstandingBalance: `${symbol || ''}${outstandingBalance || ''}`,
        storeName: String(storeName),
    };

    const mandateText = language.translate(mandateTextTranslationKey, mandateTextTranslationProps);

    useEffect(() => {
        updateMandateText(mandateText);
    }, [mandateText, updateMandateText]);

    const handleChange = (value: boolean) => {
        paymentForm.setFieldValue('orderConsent', value);
    };

    return (
        <div className="mandate-text" data-test="mandate-text">
            <CheckboxFormField
                labelContent={mandateText}
                name="orderConsent"
                onChange={handleChange}
            />
        </div>
    );
};

export default BraintreeAchMandateText;
