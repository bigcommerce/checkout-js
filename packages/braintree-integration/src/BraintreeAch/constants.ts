import { FormField } from '@bigcommerce/checkout-sdk';

import { DynamicFormFieldType } from '@bigcommerce/checkout/ui';

export enum BraintreeAchFieldType {
    BusinessName = 'businessName',
    AccountType = 'accountType',
    AccountNumber = 'accountNumber',
    RoutingNumber = 'routingNumber',
    OwnershipType = 'ownershipType',
    FirstName = 'firstName',
    LastName = 'lastName',
}

export enum OwnershipTypes {
    Personal = 'Personal',
    Business = 'Business',
}

export enum AccountTypes {
    Savings = 'Savings',
    Checking = 'Checking',
}

const accountTypeOptions = [
    {
        value: AccountTypes.Savings,
        label: AccountTypes.Savings,
    },
    {
        value: AccountTypes.Checking,
        label: AccountTypes.Checking,
    },
];

export const ownershipTypeOptions = [
    {
        value: OwnershipTypes.Personal,
        label: OwnershipTypes.Personal,
    },
    {
        value: OwnershipTypes.Business,
        label: OwnershipTypes.Business,
    },
];

export const defaultBraintreeAchFormFields: FormField[] = [
    {
        name: BraintreeAchFieldType.AccountType,
        custom: false,
        id: BraintreeAchFieldType.AccountType,
        label: 'Account Type',
        required: true,
        fieldType: DynamicFormFieldType.DROPDOWM,
        options: {
            items: accountTypeOptions,
        },
    },
    {
        name: BraintreeAchFieldType.AccountNumber,
        custom: false,
        id: BraintreeAchFieldType.AccountNumber,
        label: 'Account Number',
        required: true,
        max: 9,
        min: 8,
    },
    {
        name: BraintreeAchFieldType.RoutingNumber,
        custom: false,
        id: BraintreeAchFieldType.RoutingNumber,
        label: 'Routing Number',
        required: true,
    },
    {
        name: BraintreeAchFieldType.OwnershipType,
        custom: false,
        id: BraintreeAchFieldType.OwnershipType,
        label: 'Ownership Type',
        required: true,
        fieldType: DynamicFormFieldType.DROPDOWM,
        options: {
            items: ownershipTypeOptions,
        },
    },
];

export const personalBraintreeAchFormFields: FormField[] = [
    ...defaultBraintreeAchFormFields,
    {
        name: BraintreeAchFieldType.FirstName,
        custom: false,
        id: BraintreeAchFieldType.FirstName,
        label: 'First Name',
        required: true,
    },
    {
        name: BraintreeAchFieldType.LastName,
        custom: false,
        id: BraintreeAchFieldType.LastName,
        label: 'Last Name',
        required: true,
    },
];

export const businessBraintreeAchFormFields: FormField[] = [
    ...defaultBraintreeAchFormFields,
    {
        name: BraintreeAchFieldType.BusinessName,
        custom: false,
        id: BraintreeAchFieldType.BusinessName,
        label: 'Business Name',
        required: true,
    },
];
