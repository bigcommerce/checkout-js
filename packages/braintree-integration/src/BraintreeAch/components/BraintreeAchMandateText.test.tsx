import {
    CheckoutSelectors,
    CheckoutService,
    createCheckoutService,
} from '@bigcommerce/checkout-sdk';
import { render } from '@testing-library/react';
import { Formik } from 'formik';
import { noop } from 'lodash';
import React, { FunctionComponent } from 'react';

import { createLocaleContext, LocaleContext } from '@bigcommerce/checkout/locale';
import {
    CheckoutContext,
    PaymentFormContext,
    PaymentFormService,
} from '@bigcommerce/checkout/payment-integration-api';
import {
    getCart,
    getCheckout,
    getPaymentFormServiceMock,
    getStoreConfig,
} from '@bigcommerce/checkout/test-utils';

import { AccountTypes, OwnershipTypes } from '../constants';

import BraintreeAchMandateText, { BraintreeAchMandateTextProps } from './BraintreeAchMandateText';

describe('BraintreeAchMandateText', () => {
    let BraintreeAchMandateTextTest: FunctionComponent<BraintreeAchMandateTextProps>;
    let checkoutService: CheckoutService;
    let checkoutState: CheckoutSelectors;
    let paymentForm: PaymentFormService;

    const defaultProps = {
        updateMandateText: jest.fn().mockReturnValue('mandate text'),
        isInstrumentFeatureAvailable: false,
    };

    beforeEach(() => {
        checkoutService = createCheckoutService();
        checkoutState = checkoutService.getState();
        paymentForm = getPaymentFormServiceMock();

        global.Date.now = jest.fn(() => new Date('2023-01-01T10:00:00Z').getTime());

        jest.spyOn(checkoutState.data, 'getCheckout').mockReturnValue(getCheckout());
        jest.spyOn(checkoutState.data, 'getConfig').mockReturnValue(getStoreConfig());
        jest.spyOn(checkoutState.data, 'getCart').mockReturnValue(getCart());

        BraintreeAchMandateTextTest = (props: BraintreeAchMandateTextProps) => {
            return (
                <Formik initialValues={{}} onSubmit={noop}>
                    <CheckoutContext.Provider value={{ checkoutState, checkoutService }}>
                        <LocaleContext.Provider value={createLocaleContext(getStoreConfig())}>
                            <PaymentFormContext.Provider value={{ paymentForm }}>
                                <BraintreeAchMandateText {...props} />
                            </PaymentFormContext.Provider>
                        </LocaleContext.Provider>
                    </CheckoutContext.Provider>
                </Formik>
            );
        };
    });

    it('renders mandate text checkbox with personal data label text', () => {
        const formValues = {
            paymentProviderRadio: '', // legacy prop -> added due to the PaymentFormValues interface
            ownershipType: OwnershipTypes.Personal,
            accountType: AccountTypes.Savings,
            accountNumber: '100000000',
            routingNumber: '010000015',
            businessName: '',
            firstName: 'Name',
            lastName: 'Surname',
            shouldSaveInstrument: false,
            shouldSetAsDefaultInstrument: false,
            instrumentId: '',
            orderConsent: false,
        };

        jest.spyOn(paymentForm, 'getFormValues').mockReturnValue(formValues);

        const view = render(<BraintreeAchMandateTextTest {...defaultProps} />);

        expect(view).toMatchSnapshot();
    });

    it('renders mandate text checkbox with business data label text', () => {
        const formValues = {
            paymentProviderRadio: '', // legacy prop -> added due to the PaymentFormValues interface
            ownershipType: OwnershipTypes.Business,
            accountType: AccountTypes.Savings,
            accountNumber: '100000000',
            routingNumber: '010000015',
            businessName: 'BigcommerceStore',
            firstName: '',
            lastName: '',
            shouldSaveInstrument: false,
            shouldSetAsDefaultInstrument: false,
            instrumentId: '',
            orderConsent: false,
        };

        jest.spyOn(paymentForm, 'getFormValues').mockReturnValue(formValues);

        const view = render(<BraintreeAchMandateTextTest {...defaultProps} />);

        expect(view).toMatchSnapshot();
    });

    it('renders mandate text checkbox with text for vaulting', () => {
        const formValues = {
            paymentProviderRadio: '', // legacy prop -> added due to the PaymentFormValues interface
            ownershipType: OwnershipTypes.Personal,
            accountType: AccountTypes.Savings,
            accountNumber: '100000000',
            routingNumber: '010000015',
            businessName: '',
            firstName: 'Name',
            lastName: 'Surname',
            shouldSaveInstrument: false,
            shouldSetAsDefaultInstrument: false,
            instrumentId: '',
            orderConsent: false,
        };

        jest.spyOn(paymentForm, 'getFormValues').mockReturnValue(formValues);

        const view = render(
            <BraintreeAchMandateTextTest
                isInstrumentFeatureAvailable={false}
                updateMandateText={defaultProps.updateMandateText}
            />,
        );

        expect(view).toMatchSnapshot();
    });
});
