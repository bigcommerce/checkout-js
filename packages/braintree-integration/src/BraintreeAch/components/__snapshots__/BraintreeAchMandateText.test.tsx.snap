// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BraintreeAchMandateText renders mandate text checkbox with business data label text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mandate-text"
        data-test="mandate-text"
      >
        <div
          class="form-field"
        >
          <input
            class="form-checkbox optimizedCheckout-form-checkbox"
            id="orderConsent"
            name="orderConsent"
            type="checkbox"
            value=""
          />
          <label
            class="form-label optimizedCheckout-form-label"
            for="orderConsent"
          >
            By clicking Place Order, I authorize Braintree, a service of PayPal, on behalf of s1504098821 to verify my bank account information using bank information and consumer reports and I authorize s1504098821 to initiate an ACH/electronic debit to my savings account, Depository Name: BigcommerceStore, Routing Number: 010000015 and Account Number: 100000000, in the amount of $190 on 01.01.2023. I agree the ACH transactions I authorize comply with all applicable laws.
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mandate-text"
      data-test="mandate-text"
    >
      <div
        class="form-field"
      >
        <input
          class="form-checkbox optimizedCheckout-form-checkbox"
          id="orderConsent"
          name="orderConsent"
          type="checkbox"
          value=""
        />
        <label
          class="form-label optimizedCheckout-form-label"
          for="orderConsent"
        >
          By clicking Place Order, I authorize Braintree, a service of PayPal, on behalf of s1504098821 to verify my bank account information using bank information and consumer reports and I authorize s1504098821 to initiate an ACH/electronic debit to my savings account, Depository Name: BigcommerceStore, Routing Number: 010000015 and Account Number: 100000000, in the amount of $190 on 01.01.2023. I agree the ACH transactions I authorize comply with all applicable laws.
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BraintreeAchMandateText renders mandate text checkbox with personal data label text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mandate-text"
        data-test="mandate-text"
      >
        <div
          class="form-field"
        >
          <input
            class="form-checkbox optimizedCheckout-form-checkbox"
            id="orderConsent"
            name="orderConsent"
            type="checkbox"
            value=""
          />
          <label
            class="form-label optimizedCheckout-form-label"
            for="orderConsent"
          >
            By clicking Place Order, I authorize Braintree, a service of PayPal, on behalf of s1504098821 to verify my bank account information using bank information and consumer reports and I authorize s1504098821 to initiate an ACH/electronic debit to my savings account, Depository Name: Name Surname, Routing Number: 010000015 and Account Number: 100000000, in the amount of $190 on 01.01.2023. I agree the ACH transactions I authorize comply with all applicable laws.
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mandate-text"
      data-test="mandate-text"
    >
      <div
        class="form-field"
      >
        <input
          class="form-checkbox optimizedCheckout-form-checkbox"
          id="orderConsent"
          name="orderConsent"
          type="checkbox"
          value=""
        />
        <label
          class="form-label optimizedCheckout-form-label"
          for="orderConsent"
        >
          By clicking Place Order, I authorize Braintree, a service of PayPal, on behalf of s1504098821 to verify my bank account information using bank information and consumer reports and I authorize s1504098821 to initiate an ACH/electronic debit to my savings account, Depository Name: Name Surname, Routing Number: 010000015 and Account Number: 100000000, in the amount of $190 on 01.01.2023. I agree the ACH transactions I authorize comply with all applicable laws.
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BraintreeAchMandateText renders mandate text checkbox with text for vaulting 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mandate-text"
        data-test="mandate-text"
      >
        <div
          class="form-field"
        >
          <input
            class="form-checkbox optimizedCheckout-form-checkbox"
            id="orderConsent"
            name="orderConsent"
            type="checkbox"
            value=""
          />
          <label
            class="form-label optimizedCheckout-form-label"
            for="orderConsent"
          >
            By clicking Place Order, I authorize Braintree, a service of PayPal, on behalf of s1504098821 to verify my bank account information using bank information and consumer reports and I authorize s1504098821 to initiate an ACH/electronic debit to my savings account, Depository Name: Name Surname, Routing Number: 010000015 and Account Number: 100000000, in the amount of $190 on 01.01.2023. I agree the ACH transactions I authorize comply with all applicable laws.
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mandate-text"
      data-test="mandate-text"
    >
      <div
        class="form-field"
      >
        <input
          class="form-checkbox optimizedCheckout-form-checkbox"
          id="orderConsent"
          name="orderConsent"
          type="checkbox"
          value=""
        />
        <label
          class="form-label optimizedCheckout-form-label"
          for="orderConsent"
        >
          By clicking Place Order, I authorize Braintree, a service of PayPal, on behalf of s1504098821 to verify my bank account information using bank information and consumer reports and I authorize s1504098821 to initiate an ACH/electronic debit to my savings account, Depository Name: Name Surname, Routing Number: 010000015 and Account Number: 100000000, in the amount of $190 on 01.01.2023. I agree the ACH transactions I authorize comply with all applicable laws.
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
